### ISSUE TITLE
Msgs not going to Kafka

### ISSUE THREAD

**Issue by crashinc99** [ipa-digitools/simple-kafka-mqtt-connector#3]
Hi,

  I'm having trouble getting messages from mqtt-connector to kafka.

  I'm seeing messages appear in the logs from the MQTT server and attempts to write to kafka:

[2020-05-13 19:13:51,451] (org.apache.kafka.clients.NetworkClient) - WARN [Producer clientId=simple-kafka-mqtt-connector-1] Connection to node 1008 (kafka/172.18.0.3:29092) could not be established. Broker may not be available. 
[2020-05-13 19:13:52,410] (org.apache.kafka.clients.NetworkClient) - WARN [Producer clientId=simple-kafka-mqtt-connector-1] Connection to node 1008 (kafka/172.18.0.3:29092) could not be established. Broker may not be available. 
[2020-05-13 19:13:53,268] (org.apache.kafka.clients.NetworkClient) - WARN [Producer clientId=simple-kafka-mqtt-connector-1] Connection to node 1008 (kafka/172.18.0.3:29092) could not be established. Broker may not be available. 
[2020-05-13 19:15:43,546] (de.fhg.ipa.null70.simple_kafka_mqtt_connector.SimpleKafkaMQTTConnector) - INFO andf99-topic - wtf7 
[2020-05-13 19:15:43,560] (org.apache.kafka.clients.NetworkClient) - WARN [Producer clientId=simple-kafka-mqtt-connector-1] 1 partitions have leader brokers without a matching listener, including [replication-in-0] 
[2020-05-13 19:15:43,661] (org.apache.kafka.clients.NetworkClient) - WARN [Producer clientId=simple-kafka-mqtt-connector-1] 1 partitions have leader brokers without a matching listener, including [replication-in-0] 

Does mqtt-connector attempt to connect to kafka on startup?  I don't see connection errors til it receives a message from the mqtt server.

  I'm using docker-compose to try and get this working.  Config:

  kafka:
    image: wurstmeister/kafka:2.11-2.0.0
    ports:
      - "9092:9092"
      - "29094:29094"
    expose:
      - "29092"
    environment:
      KAFKA_LISTENERS: BOB://0.0.0.0:29092, FRED://0.0.0.0:9092, ALICE://0.0.0.0:29094
      KAFKA_ADVERTISED_LISTENERS: BOB://kafka:29092,FRED://localhost:9092,ALICE://192.168.1.98:29094
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: BOB:PLAINTEXT, FRED:PLAINTEXT, ALICE:PLAINTEXT
      KAFKA_INTER_BROKER_LISTENER_NAME: BOB
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_AUTO_CREATE_TOPICS_ENABLE: 'true'

#######  
#
#   simple-kafka-mqtt-connector
#   
########
      
# !EDITME! IP THROUGH WHICH THE KAFKA BROKER WILL BE REACHED
#KAFKA_HOST=kafka
KAFKA_HOST=kafka
      
KAFKA_PORT=29092
KAFKA_CLIENT_ID=simple-kafka-mqtt-connector-1
      
# !EDITME! IP THROUGH WHICH THE MQTT BROKER WILL BE REACHED
MQTT_HOST=broker.hivemq.com
MQTT_PORT=1883
MQTT_CLIENT_ID=simple-kafka-mqtt-connector-1
MQTT_QOS=2
      
#!EDITME! mqttTopicA>>>kafkaTopicA;mqttTopicB>>>kafkaTopicB;mqttTopicC>>>kafkaTopicC
#TOPIC_MAPPING=robotgroup001/robot001>>>test;robotgroup001/robot002>>>test02;robotgroup001/robot003>>>test03
TOPIC_MAPPING=andf99-topic>>>replication-in
~                                                                              
Thanks,
Andy

**Comment by crashinc99** [comment]
I fixed my Kafka issues.  I'm seeing my MQTT message appear in the docker log for the simple-kafka-mqtt-connector.

I'm not seeing anything show up in kafka though.

Should I see audits that indicate a message was written to a kafka queue?

Also, does simple-kafka-mqtt-connector make use of the kafka group id?

Thanks,
Andy

**Comment by AGRG-IPA-070** [comment]
Is auto.create.topics.enable true for your kafka broker?

**Comment by crashinc99** [comment]
Yes it is.  I got things working but not sure how.  I messed around with the kafka queues.  At one point I built them manually in one kafka instance.  That got rid of errors but no messages were appearing.  I ended up reverting to a different kafka instance, renamed topics, and mqtt messages started appearing.

I didn't see any log entries for successful mqtt -> kafka writes so I'm relying on my kafka client to see the messages.

Does your software provide kafka -> mqtt bridging as well?  

Thanks for your time!
